### PR TITLE
Added check for "https" in sysmeta update functions

### LINKED ISSUE #47: Modify rights/access functions to prevent use of `https` ORCID URIs
Functions like `set_rights_and_access` can be passed any `Subject` and a common mistake is to pass in an HTTPS ORCID URI for a `Subject`. This is a problem because DataONE only uses the HTTP form and doesn't consider the HTTPS variant to be equivalent. So, currently, the user has to be careful to not to use the HTTPS form.

The various functions that set `rightsHolder` and `accessPolicy` information on SysMeta could easily check for this specific case.

### PR DESCRIPTION
**Pull request by dmullen17**
resolve #47  Came up with 3 potential fixes, chose the first to keep the code style consistent.  The other two are included in case you prefer one.  They would need to be amended for the multiple subjects case in `set_access`

if (grepl("https", subject)) {
  stop("https format for subject not accepted, change to http")
}

if(grepl("https", subject)){
  subject <- gsub("https", "http", subject)
}

### CODE REVIEW

**Comment by amoeba** [comment]
Thanks for PR'ing this!

I think I like `stop`ing, instead of fixing, for two reasons:

- Helps train the team member that they've done something wrong and how to fix it
- Avoids making a choice for them they might not want

I see in your examples above, your `stop` condition is when `https:` is found in the subject but that, in the commits you've put in this PR, your stop condition is when `http:` is found. A few things:

- I think the code should make their parent functions stop when a subject is passed in with "https:", not "http:"
- `grepl` returns a logical vector of the same size as the `x` argument so the `if` statement, as written, should not quite work as expected and also produce a warning. I suggest wrapping it inside a call to `any()`.
- I've used `stopifnot` a lot in my code but I've realized that it makes for hard to understand error messages for the team so I've tried to write more helpful messages, aimed at a hypothetical new data team member. Could you add that too?
- `grepl`, and related functions, support regular expressions so the `grepl` call can be more specific. To catch subjects that *start* with `https:`, we'd write `grepl("^https:", x)` and we can be even more specific by using `grepl("^https:\\/\\/orcid\\.org", x)`

What do you think about all that?

**Comment by dmullen17** [comment]
That all makes sense.  Thanks for the thoughtful response!  

- I only need to wrap `grepl` inside a call to `any` for the possible multiple subjects input in `set_access` right?  

- When you mention "could you add that too" would you like me to update all the initial `stopifnot` calls to `if(..){stop(..)}` format?  Most of the `stopifnot` error messages make enough sense, like `is.character(pids) is not TRUE`, even if they don't explicitly spell out the error in layman's terms.  But I'd be happy to update those if you'd like.  

- `grepl("^https:\\/\\/orcid\\.org", x)` is much cleaner, I like it!

**Comment by amoeba** [comment]
1. Yes
2. Yes please. You can just add more commits to your fork and this PR will update. From a perspective of someone who's a relatively experienced but once inexperienced data team member, do you think it's worth the time to write out such helpful error messages?
3. 👍

**Comment by dmullen17** [comment]
Ok sounds good.  It's probably not worth updating every function, but I'll add the messages for `publish_update` and `publish_object` - the two most frequently used functions (at least in my experience).

**Comment by amoeba** [comment]
I don't totally follow what the update to those two functions would be. I was thinking you'd just modify the relevant functions in `access.R`. What kind of changes were you thinking about?

**Comment by dmullen17** [comment]
Just changing the initial `stopifnot` calls to produce error messages.  For instance lines 247-276 in `editing.R`.  But I'm happy to stick to modifying the relevant functions in `access.R`

**Comment by amoeba** [comment]
Ah, gotcha. For those, I'd appreciate a separate PR just to keep things simple.

**Comment by amoeba** [comment]
PS: When you get to it, please add yourself as a contributor in the DESCRIPTION file.

**Comment by dmullen17** [comment]
Understood, I'm still getting up to speed with conventions, thanks for the patience / detailed explanations!

**Comment by amoeba** [comment]
Cool, those changes look good. Can you please add yourself as a contributor in the `DESCRIPTION` file and then I'll merge this?

**Comment by amoeba** [comment]
Nevermind on ^^! You're already listed aren't you?

**Comment by dmullen17** [comment]
Hey @amoeba I just ran into an unintended consequence of this fix.  If an orcid with a `https` URI is already in the system metadata it's difficult to update/remove it because `set_rights_and_access` (and other related functions) won't accept it as an argument input.

What do you think about changing the `stop` calls to `warning` and having the functions change `https` to `http`?  We'd originally agreed `stop` was better for the following reasons: 
```
- Helps train the team member that they've done something wrong and how to fix it
- Avoids making a choice for them they might not want
```
Now I'm not sure it's better as it's pretty difficult to work with packages that contain `https` URI's in the sysmeta.  I'd be happy to PR this if you want to go this route.  
Alternatively we could query all the ADC objects and convert `https` to `http` all at once (not sure how sane this idea is)!
